spec/reservation_maker_spec.rb

```diff
@@ -0,0 +1,162 @@
+    it "collates all reservations into master @@reservations array" do
+
+      start_date1 = Date.new(2018,2,3)
+      end_date1 = Date.new(2018,2,5)
+      ReservationMaker.new(start_date1, end_date1).create_reservation
+      # binding.pry
+
+      start_date2 = Date.new(2018,3,1)
+      end_date2 = Date.new(2018,3,2)
+      ReservationMaker.new(start_date2, end_date2).create_reservation
+      # binding.pry
+
+      expect(ReservationMaker.reservations.length).must_equal 2
+
+    end
+  end
+
+  describe "tests for StandardError if no rooms are available. " do
+    it "it will throw an error if all rooms are already booked" do
+
+      available_rooms = nil
+
+      start_date1 = Date.new(2018,2,3)
+      end_date1 = Date.new(2018,2,5)
+      example1 = ReservationMaker.new(start_date1, end_date1).create_reservation
+
+      expect(example1).must_raise StandardError
+
+    end
+  end
+
+  describe "tests for correctly making reservations. " do
+    it "it will add all 20 reservations booked for the same date" do
+
+      20.times do
+        ReservationMaker.new(Date.new(2018,2,3), Date.new(2018,2,5)).create_reservation
+      end
+
+      expect(ReservationMaker.reservations.length).must_equal 20
+
+    end
+  end
+
+  describe "tests whether another reservation will be made if all rooms are booked. " do
+    before do
+
+      20.times do
+        ReservationMaker.new(Date.new(2018,2,3), Date.new(2018,2,5)).create_reservation
+      end
+    end
+    it "it will NOT create another reservation because the 21st reservation is for the SAME DATES as the first 20 reservations." do
+
+      ReservationMaker.new(Date.new(2018,2,3), Date.new(2018,2,5)).create_reservation
+
+      expect(ReservationMaker.reservations.length).must_equal 20
+
+    end
+
+    it "it will NOT create another reservation because the 21st reservation OVERLAPS with the START dates of the first 20 reservations." do
+
+      ReservationMaker.new(Date.new(2018,2,3), Date.new(2018,2,8)).create_reservation
+
+      expect(ReservationMaker.reservations.length).must_equal 20
+
+    end
+
+    it "it will create another reservation because the 21st reservation STARTS ON THE CHECK-OUT dates of the first 20 reservations." do
+
+      ReservationMaker.new(Date.new(2018,2,5), Date.new(2018,2,8)).create_reservation
+
+      expect(ReservationMaker.reservations.length).must_equal 21
+
+    end
+
+    it "it WILL create another reservation because the 21st reservation starts on the check-out dates of the first 20 reservations." do
+
+      ReservationMaker.new(Date.new(2018,2,5), Date.new(2018,2,8)).create_reservation
+
+      expect(ReservationMaker.reservations.length).must_equal 21
+
+    end
+
+    it "it will NOT create another reservation because the 21st reservation is for dates contained within the start and end dates of the first 20 reservations." do
+
+      example = ReservationMaker.new(Date.new(2018,2,4), Date.new(2018,2,5))
```

Comment:
The local variable `example` ends up not being used in the rest of this method-- do we need it?